### PR TITLE
Add utility scripts to streamline build and testing process

### PR DESCRIPTION
**Pull request by spicykimche**

### CODE REVIEW

**Comment by spicykimche** [comment]
I made the README sound as generic as possible, in case you wanted to let other students add onto it.

File: decaflex/contrib/evaluate.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FILES=references/dev/*.out
+for filename in $FILES; do 
+	IFS='/' read -r -a file <<< "$filename" 
+	REFERENCE=references/dev/${file[2]} 
+	OUTPUT=output/dev/${file[2]} 
+	DIFF=$(diff $REFERENCE $OUTPUT)
```
Comment (anoopsarkar) [inline decaflex/contrib/evaluate.sh]: might be better to use `diff -y --suppress-common-lines` to see the output against references side by side.

File: decaflex/contrib/evaluate.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FILES=references/dev/*.out
```
Comment (jryzkns) [inline decaflex/contrib/evaluate.sh:1]: This script does not check for `.ret` files for the `e*.decaf` test cases, please consider adding this functionality

File: decaflex/contrib/evaluate.sh (hunk 1 above)
Comment (jryzkns) [inline decaflex/contrib/evaluate.sh]: please add `-w` option to the diff command so the output of this file matches the outputs in `check.py`

File: decaflex/contrib/evaluate.sh (hunk 2 above)
Comment (spicykimche) [reply decaflex/contrib/evaluate.sh:1]: In the next PR, I will add functionality to check *.ret files as well. Thanks for the heads up.

**Review by anoopsarkar — COMMENTED** [review]
The `result.sh` needs a filename to show you the input, expected output and the answer program output. It might be hard for people to figure out which files they should be looking at. One way to make this easier is to put a `while read f` loop in `result.sh` so it reads a list of filenames and then in `evaluate.sh` you can change lines 9 and 10 so that the `echo` of the filenames are sent to stderr using `echo $DIFF 1>&2` and only producing the file names (omit the `no discrepancies`). 

This way you can view the diffs while running `evaluate.sh` but also pipe the output directly to `result.sh` like so:

`bash evaluate.sh | xargs result.sh`

Then it will show you the file comparison for only the cases where there was a non-empty diff.

File: decaflex/contrib/evaluate.sh (hunk 1 above)
Comment (anoopsarkar) [reply decaflex/contrib/evaluate.sh]: In my above comment I mistakenly referred to using `while read f` in `result.sh`. But that reads from stdin not the command line argument list, which should be done like this:
```
# Reading argument values using loop
for argval in "$@"
do
  echo -n "$argval  "
done
```

**Comment by anoopsarkar** [comment]
I'll merge it once the review comments have been resolved. thanks!

**Comment by spicykimche** [comment]
> I'll merge it once the review comments have been resolved. thanks!

All changes have been addressed. Ready for review.

**Review by anoopsarkar — APPROVED** [review]

**Comment by anoopsarkar** [comment]
PR is merged. Please describe the contents of the contrib dir in a coursys discussion board post and tell them how they can get the update.

**Pull request merged**